build_test/lib/src/resolve_source.dart

```diff
@@ -69,16 +86,18 @@ Future<Resolver> resolveSource(
 /// input given a set of dependencies to also use. See `resolveSource`.
 class _ResolveSourceBuilder implements Builder {
   final Completer<Resolver> _resolver;
+  final Future<Null> _tearDown;
 
-  const _ResolveSourceBuilder(this._resolver);
+  const _ResolveSourceBuilder(this._resolver, this._tearDown);
 
   @override
-  Future<Null> build(BuildStep buildStep) {
-    return new Future.sync(() {
+  Future<Null> build(BuildStep buildStep) async {
+    await new Future.sync(() {
```

Comment:
Now that this is an `async` method you should be able to remove this explicit Future?